libtransact/src/protocol/batch.rs

```diff
@@ -142,6 +142,10 @@ impl Batch {
         &self.header
     }
 
+    pub fn deserialized_header(&self) -> Result<BatchHeader, BatchBuildError> {
```

Comment:
I don't think I agree with this change - `BatchPair` and `TransactionPair` is the intended route as it essentially caches the header deserialization.  This function gives the user a route to get the header, but doesn't cache it, so it could create an unexpected frequent expense.

Reply:
In libsawtooth/sawtooth-core, you're mostly dealing with references to batches and transactions because you're getting them from a block. This means to get the header, you have to clone the whole thing then build it; and that's usually just to get a single field from the header. That's not efficient or ergonomic. It seems there's value in having both options available; we could add a note in a doc string for this method to provide information on when you would use this over `into_pair`.

Reply:
Removed this commit